csrc/core/math.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+template <typename A, typename B>
+static inline constexpr auto div_ceil(A a, B b) {
+  return (a + b - 1) / b;
+}
+
+// Compute the next multiple of a that is greater than or equal to b
+template <typename A, typename B>
+static inline constexpr auto next_multiple_of(A a, B b) {
+  return div_ceil(b, a) * a;
+}
+
+// Compute the largest multiple of a that is less than or equal to b
+template <typename A, typename B>
+static inline constexpr auto prev_multiple_of(A a, B b) {
+  return (b / a) * a;
+}
```

Comment:
I got a little confused about what these functions were doing and thought you would want to do something like this instead:
```
template <typename T>
inline constexpr T round_to_previous_multiple_of(T a, type_wrapper_t<T> b)
{
    detail::check_all_nonzero(b);
    return a % b == 0 ? a : (a / b) * b;
}

template <typename T>
inline constexpr T round_to_next_multiple_of(T a, type_wrapper_t<T> b)
{
    detail::check_all_nonzero(b);
    return a % b == 0 ? a : ((a / b) + 1) * b;
}
```